### PR TITLE
Don't use sub-path twice for manifest

### LINKED ISSUE #14422: Chrome tries to load .../gitea/gitea/img/logo.png, Firefox loads .../gitea/img/logo.png
- Gitea version (or commit ref): 1.14.0+dev-575-g1c230f69d 
- Git version: 2.25.1
- Operating system: ubuntu-20.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
- I'm using the context /gitea for my instance

## Description

After clicking on the upper left logo, chrome tries to load several urls. There are two logo urls:

- logo.svg ... this is ok - https://hetzner-de.daemons-point.com/gitea/img/logo.svg
- logo.png ... this is KO, it contains the context "/gitea" twice - https://hetzner-de.daemons-point.com/gitea/gitea/img/logo.png

For firefox the error doesn't happen. Firefox does not try to load the PNG logo.

## Screenshots

Chrome:

image

Firefox:

image

### PR DESCRIPTION
**Pull request by uli-heller**
Fixes #14422

### CODE REVIEW

**Comment by techknowlogick** [comment]
Thanks for this PR!

It is failing on the CI due to needing to run `make fmt` on your local working copy and pushing the results of that.

File: modules/setting/setting.go [hunk 1]
```diff
@@ -79,32 +79,33 @@ var (
...
+	PprofDataPath           string
+	EnableLetsEncrypt       bool
+	LetsEncryptTOS          bool
+	LetsEncryptDirectory    string
+	LetsEncryptEmail        string
+	GracefulRestartable     bool
+	GracefulHammerTime      time.Duration
+	StartupTimeout          time.Duration
+	StaticURLPrefix         string
+	StaticURLPrefixManifest string
```
Comment (6543) [inline modules/setting/setting.go]: I'm not sure, adding a special var only for manifest is the right way to fix it

Comment (uli-heller) [reply modules/setting/setting.go]: Feel free to change it! I do not know anything about go and my heart isn't connected with this pull request. All I know is that it fixes the issue for me. And I know it is not very nicely done. So: You surely don't offend me if you fix it differently

Comment (6543) [reply modules/setting/setting.go]: First thanks for the pull, I think we can evolve your pull ;) together ...

**Comment by 6543** [comment]
I would wait until #14293 got merged ...

**Comment by uli-heller** [comment]
Does someone know why the fix only works when using this ordering:

```
// OK - manifest contains correct urls
AbsoluteAssetURL = MakeAbsoluteAssetURL(AppURL, strings.TrimSuffix(sec.Key("STATIC_URL_PREFIX").MustString(""), "/"))
StaticURLPrefix = strings.TrimSuffix(sec.Key("STATIC_URL_PREFIX").MustString(AppSubURL), "/")
```

When I change the ordering, the manifest contains the wrong urls, those based on StaticURLPrefix:

```
// KO - manifest contains wrong urls based on StaticURLPrefix
StaticURLPrefix = strings.TrimSuffix(sec.Key("STATIC_URL_PREFIX").MustString(AppSubURL), "/")
AbsoluteAssetURL = MakeAbsoluteAssetURL(AppURL, strings.TrimSuffix(sec.Key("STATIC_URL_PREFIX").MustString(""), "/"))
```

**Comment by silverwind** [comment]
Hmm if this is really fixed by reordering statements, there must be another underlying issue that causes the variable to go wrong somewhere else. It should only be assigned once.

**Comment by uli-heller** [comment]
Sorry, maybe my explanation is misleading.

The issue is caused by this:

```
StaticURLPrefix = strings.TrimSuffix(sec.Key("STATIC_URL_PREFIX").MustString(AppSubURL), "/")
```

It makes "StaticURLPrefix" an URL that contains the sub-path twice, i.e. "https://mygitea.tld/gitea/gitea".
`strings.TrimSuffix(sec.Key("STATIC_URL_PREFIX").MustString(""), "/")` fixes this.

But only it is it called before `StaticURLPrefix=strings.TrimSuffix(sec.Key("STATIC_URL_PREFIX").MustString("AppSubURL"), "/")`. Currently, I don't understand why it has to be called first...

**Comment by 6543** [comment]
could be #12999 an alternative for this pull?!?

**Comment by uli-heller** [comment]
@6543 : Shall I give it - #12999 - a try and report if it fixes the issue?

**Comment by 6543** [comment]
@uli-heller would be nice ... but we first have to wait to see conflicts resolved ...

**Comment by uli-heller** [comment]
@6543 : Merging worked without an issue, building too. Unfortuntely, the issue is not fixed by #12999:
image

**Comment by 6543** [comment]
thanks for your work & testing.

I'll close this since #14827 did catch the real issue behind it :)

**Comment by uli-heller** [comment]
@6543  : Thx a lot!

**Pull request closed without merging**